data/lib/src/network/linshare_http_client.dart

```diff
@@ -128,6 +131,35 @@ class LinShareHttpClient {
     return resultJson.map((data) => _getDynamicAutoCompleteResult(data)).toList();
   }
 
+  Future<List<WorkGroupNodeDto>> getWorkGroupChildNodes(
+      SharedSpaceId sharedSpaceId,
+      {WorkGroupNodeId parentId}
+  ) async {
+    final endpointPath = Endpoint.sharedSpaces
+        .withPathParameter('/${sharedSpaceId.uuid}')
+        .withPathParameter(Endpoint.nodes)
+        .generateEndpointPath();
+
+    final queryParameters = parentId != null
+        ? [StringQueryParameter('parent', parentId.uuid)]
+        : <QueryParameter>[];
+
+    final List nodesJsonResult = await _dioClient.get(
+      endpointPath,
+      queryParameters: queryParameters.toMap(),
+    );
+
+    // Sort by modificationDate descending
+    final sortedWorkgroupNodesDto = nodesJsonResult
```

Comment:
I think you should sort the list in datasource, since this class should only return raw data instead of extra logic code.